src/Cryptol/Eval/Reference.lhs

```diff
@@ -1284,8 +1297,10 @@ False]`, but to `error "foo"`.
 >           do val' <- val
 >              pure $ VRecord [ (f, go fty (lookupRecord f val'))
 >                             | (f, fty) <- canonicalFields fs ]
->         TVNominal {} ->
->           evalPanic "arithUnary" ["Nominal type not in `Ring`"]
+>         TVNominal _ _ ntv ->
+>           case ntv of
+>             TVStruct fs -> go (TVRec fs) val
+>             _ -> evalPanic "arithUnary" ["Enum/Abstract type not in `Ring`"]
```

Comment:
Similarly, this (and the other `evalPanic` cases) should mention `"ringUnary"` instead of `"arithUnary"` here.